File: easy/palindrome-check/typescript/isPalindrome.ts
export const isPalindrome = (s: string): boolean => {
  const sanitized = s.replace(/[^a-zA-Z0-9]/g, '').toLowerCase();
  let left = 0;
  let right = sanitized.length - 1;

  while (left < right) {
    if (sanitized[left] !== sanitized[right]) {
      return false;
    }
    left++;
    right--;
  }

  return true;
};

File: easy/palindrome-check/csharp/isPalindrome.cs
using System.Text.RegularExpressions;

public class Solution {
    public bool IsPalindrome(string s) {
        string sanitized = Regex.Replace(s, "[^a-zA-Z0-9]", "").ToLower();
        int left = 0;
        int right = sanitized.Length - 1;

        while (left < right) {
            if (sanitized[left] != sanitized[right]) {
                return false;
            }
            left++;
            right--;
        }

        return true;
    }
}
